tree_test.go

```diff
@@ -285,37 +286,46 @@ func TestDeletePrune(t *testing.T) {
 	key2, _ := hex.DecodeString("0107000000000000000000000000000000000000000000000000000000000000")
```

Comment:
This is a very useful test, I think I never looked at it before.
The essence is the same, but has two extra nits:
- Adds some extra inserted key (`key5`) that puts a value in the same leaf as (`key4`). This is to also cover C2 updating and not only C1 updating.
- Added some extra comments to simply help readers understand what we're doing here. It might be surprising why the hash _shouldn't_ match, which is actually the right thing to happen (because of the "leaf marker").